src/java/com/unifina/signalpath/blockchain/SolidityCompileDeploy.java

```diff
@@ -165,7 +164,7 @@ protected BigInteger getGasLimit() {
 	}
 
 	protected Web3j getWeb3j() {
-		return Web3j.build(new HttpService(ethereumOptions.getRpcUrl()));
+		return ethereumOptions.getWeb3j(EthereumModuleOptions.RpcConnectionMethod.HTTP);
```

Comment:
is there a specific reason for this to be HTTP?

Reply:
just didn't want to make any breaking changes. I can test WS and switch if you want. I'm not sure how web3j WS handles reconnection.